pkg/chunk/purger/blocks_purger.go

```diff
@@ -0,0 +1,324 @@
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+func (api *BlocksPurgerAPI) GetAllDeleteRequestsHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+	tManager := cortex_tsdb.NewTombstoneManager(api.bucketClient, userID, api.cfgProvider, api.logger)
+	deleteRequests, err := tManager.GetAllDeleteRequestsForUser(ctx)
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "error getting delete requests from the block store", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(deleteRequests); err != nil {
+		level.Error(util_log.Logger).Log("msg", "error marshalling response", "err", err)
+		http.Error(w, fmt.Sprintf("Error marshalling response: %v", err), http.StatusInternalServerError)
+	}
+
+	w.WriteHeader(http.StatusOK)
+
+}
+
+func (api *BlocksPurgerAPI) CancelDeleteRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	userID, err := tenant.TenantID(ctx)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	params := r.URL.Query()
+	requestID := params.Get("request_id")
+
+	tManager := cortex_tsdb.NewTombstoneManager(api.bucketClient, userID, api.cfgProvider, api.logger)
+	deleteRequest, err := tManager.GetDeleteRequestByIDForUser(ctx, requestID)
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "error getting delete request from the object store", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if deleteRequest == nil {
+		http.Error(w, "could not find delete request with given id", http.StatusBadRequest)
+		return
+	}
+
+	if deleteRequest.State == cortex_tsdb.StateCancelled {
+		http.Error(w, "the request has already been previously deleted", http.StatusBadRequest)
+		return
+	}
+
+	if deleteRequest.State == cortex_tsdb.StateProcessed {
+		http.Error(w, "deletion of request which is already processed is not allowed", http.StatusBadRequest)
+		return
+	}
+
+	if time.Since(deleteRequest.GetCreateTime()) > api.deleteRequestCancelPeriod {
+		http.Error(w, fmt.Sprintf("Cancellation of request past the deadline of %s since its creation is not allowed", api.deleteRequestCancelPeriod.String()), http.StatusBadRequest)
+		return
+	}
+
+	// create file with the cancelled state
+	_, err = tManager.UpdateTombstoneState(ctx, deleteRequest, cortex_tsdb.StateCancelled)
+	if err != nil {
+		level.Error(util_log.Logger).Log("msg", "error cancelling the delete request", "err", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+// Calculates the tombstone file name based on a hash of the start time, end time and selectors
+func getTombstoneRequestID(startTime int64, endTime int64, selectors []string) string {
```

Comment:
I prefer to have functions named after what they do, not after how you expect to use them.
So this can be `getTombstoneHash`; the current function with that name doesn't need to exist.

Reply:
Makes total sense. Will keep that in mind going forward, thanks.